src/compute/src/render/join/linear_join.rs

```diff
@@ -509,12 +573,13 @@ where
         for<'a> Tr1::Val<'a>: IntoRowByTypes,
         for<'a> Tr2::Val<'a>: IntoRowByTypes,
     {
+        let join_spec = LinearJoinSpec::from_config(&self.worker_config);
```

Comment:
We could also compute this only when the ConfigSet changes (i.e. in `handle_update_configuration`) and continue to store it directly on the worker config, if the perf of all the parsing matters

Reply:
Good call! Changed.